### PR TITLE
fix: invalidating history on wallet reload

### PR DESCRIPTION
**Pull request by andreabadesso**
### Acceptance Criteria
- We should invalidate the token history after a connection reload so we make sure we didn't lose transactions

### Security Checklist
- [X] Make sure you do not include new dependencies in the project unless strictly necessary and do not include dev-dependencies as production ones. More dependencies increase the possibility of one of them being hijacked and affecting us.

### CODE REVIEW

**Review by r4mmer — APPROVED** [review]

File: src/sagas/wallet.js [hunk 1]
```diff
@@ -502,10 +508,42 @@ export function* onWalletReloadData() {
...
+  // Wait until the wallet is ready
+  yield take(types.WALLET_STATE_READY);
+
+  try {
+    const registeredTokens = yield call(loadTokens);
+
+    const customTokenUid = DEFAULT_TOKEN.uid;
+    const htrUid = hathorLibConstants.HATHOR_TOKEN_CONFIG.uid;
+    // We might have lost transactions during the reload, so we must invalidate the
+    // token histories:
```
Comment (pedroferreira1) [inline src/sagas/wallet.js:527]: I would just add a comment here explaining why you skip the invalidation for HTR and default token. As far as I understood, the other reload wallet method already handles balance/history for them, right?

Also the names seem a bit confusing to me:

`onWalletReloadData` and `walletReloading`

Maybe this new one should be called `beforeWalletReload`. What do you think?

Comment (andreabadesso) [reply src/sagas/wallet.js:513]: Actually, we should activate this saga when `reload` is emitted from the facade as well. 

Refactored this in f7f70a20861de3573f1141cd21201b8aa02fa115

@r4mmer can you please review again?

File: src/sagas/wallet.js [hunk 2]
```diff
@@ -543,11 +530,11 @@ export function* saga() {
   yield all([
     takeLatest('START_WALLET_REQUESTED', startWallet),
     takeLatest('WALLET_CONN_STATE_UPDATE', onWalletConnStateUpdate),
-    takeLatest('WALLET_RELOADING', walletReloading),
+    takeLatest('WALLET_RELOADING', onWalletReloadData),
+    takeLatest('WALLET_RELOAD_DATA', onWalletReloadData),
```
Comment (r4mmer) [inline src/sagas/wallet.js:534]: Didn't we remove this action?

Comment (andreabadesso) [reply src/sagas/wallet.js:534]: Fixed in ca011434a77409fb5a8b22e73f8c3f66dd03e7e6

**Review by pedroferreira1 — APPROVED** [review]

**Review by pedroferreira1 — APPROVED** [review]

**Review by r4mmer — APPROVED** [review]

**Pull request merged**